src/main/java/io/github/thebusybiscuit/slimefun4/core/multiblocks/MultiBlockMachine.java

```diff
@@ -104,12 +104,20 @@ public void postRegister() {
     @Override
     public void load() {
         super.load();
-
-        for (ItemStack recipeItem : displayRecipes) {
-            SlimefunItem item = SlimefunItem.getByItem(recipeItem);
-
-            if (item == null || !item.isDisabled()) {
-                recipes.add(new ItemStack[] { recipeItem });
+        
+        for (ItemStack inputStack : displayRecipes) {
+            int index = displayRecipes.indexOf(inputStack);
+            if (index % 2 != 0) {
+                continue;
+            }
```

Comment:
Might also be worth to just do `for (int i = 0; i < displayRecipes.size(); i +=2)` here to save all this stuff.